packages/nimble-components/src/wafer-map/modules/experimental/worker-renderer.ts

```diff
@@ -1,14 +1,82 @@
+import { type Remote, transfer } from 'comlink';
 import type { WaferMap } from '../..';
 import { HoverDieOpacity } from '../../types';
+import { createMatrixRenderer } from '../create-matrix-renderer';
+import type { MatrixRenderer } from '../../../../build/generate-workers/dist/esm/source/matrix-renderer';
 
 /**
  * Responsible for drawing the dies inside the wafer map, adding dieText and scaling the canvas
  */
 export class WorkerRenderer {
+    private matrixRenderer!: Remote<MatrixRenderer>;
+
     public constructor(private readonly wafermap: WaferMap) {}
 
-    public drawWafer(): void {
-        // rendering will be implemented in a future PR
+    public async setupWafer(): Promise<void> {
```

Comment:
@Razvan1928 pushed a few changes to split setup and draw wafer into different responsibilities. If it looks good ready to go!

Reply:
Took a look on the commits, everything looks great!